lib/rules/index.js

```diff
@@ -3,7 +3,7 @@
 /* eslint global-require: 0 */
 
 /** @satisfies {Record<string, import('eslint').Rule.RuleModule>} */
-module.exports = {
+const rules = {
```

Comment:
`@satisfies` doesn't work for `module.exports = {` and you have to use a variable here to make it work. That's why the missing `/** @type {import('eslint').Rule.RuleModule} */` in `lib/rules/forward-ref-uses-ref.js` wasn't be found before.